### PR TITLE
Add the ability to move a control point around in time

### PR DESCRIPTION
**Pull request by peppy**
For now, this does a reconstruction of the group, as there is no way to modify the time (I believe this was the intentional direction to keep things simple, and should be fine for the time being).

- [x] Depends on #10324

### CODE REVIEW

File: osu.Game/Screens/Edit/Timing/GroupSection.cs [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+                textBox.Current.Disabled = false;
+                textBox.Text = $"{group.NewValue.Time:n0}";
+            }, true);
+        }
+
+        private void changeSelectedGroupTime(in double time)
+        {
+            var currentGroupItems = SelectedGroup.Value.ControlPoints.ToArray();
+
+            Beatmap.Value.Beatmap.ControlPointInfo.RemoveGroup(SelectedGroup.Value);
```
Comment (smoogipoo) [inline osu.Game/Screens/Edit/Timing/GroupSection.cs:95]: So how do you envision this working from a change handler perspective? Is `EditorBeatmap` going to bind to `ControlPointInfo.Groups` and bind to `ItemAdded`/`ItemRemoved` events, then bind to events from the bindables?

Or, is section going to call something like `changeHandler.SaveState()`?

Comment (peppy) [reply osu.Game/Screens/Edit/Timing/GroupSection.cs:95]: I have a PR in the works to fix that (https://github.com/ppy/osu/issues/10314). It just saves a state when the `ControlPointInfo.Groups` `BindableList` fires a change.

Comment (peppy) [reply osu.Game/Screens/Edit/Timing/GroupSection.cs:100]: Have added the local logic required to ensure only one state saved.

Comment (smoogipoo) [reply osu.Game/Screens/Edit/Timing/GroupSection.cs:103]: It doesn't seem to want to undo the changes?

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**